cadquery/occ_impl/assembly.py

```diff
@@ -289,3 +297,167 @@ def toJSON(
+
+    :param assy: Assembly that is being converted to a compound for the document.
+    :param assy_name: Label that is applied to the top level object in the document.
+    """
+
+    # Prepare a document
+    app = XCAFApp_Application.GetApplication_s()
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+    app.InitDocument(doc)
+
+    # Shape and color tools
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+    shape_tool.SetAutoNaming_s(False)
+
+    # Convert the assembly to a compound
+    shape_list = []
+    for part in assy.children:
+        for shape in part.toShapeList():
+            # Make sure that we only add solids
+            if isinstance(shape, Solid):
+                shape_list.append(shape)
+    comp = Compound.makeCompound(shape_list).locate(part.loc).wrapped
+
+    # Add the top level shape, have it broken apart into an assembly and set the top-level name
+    top_level_lbl = shape_tool.AddShape(comp, True)
+    TDataStd_Name.Set_s(top_level_lbl, TCollection_ExtendedString(assy_name))
+
+    # Assign names and colors based on the assembly
+    for part in assy.children:
+        for shape in part.toShapeList():
+            # Make sure that we only trying to set the names and colors for solids
+            if isinstance(shape, Solid):
+                cur_lbl = shape_tool.FindShape(shape.wrapped)
+                TDataStd_Name.Set_s(cur_lbl, TCollection_ExtendedString(part.name))
+
+                # Make sure that a color was passed in
+                if part.color:
+                    color_tool.SetColor(cur_lbl, part.color.wrapped, XCAFDoc_ColorGen)
+
+    return doc
+
+
+def toFusedCAF(
+    assy: AssemblyProtocol,
+    assy_name: str,
+    glue: Optional[bool] = False,
+    tol: Optional[float] = None,
+) -> TDocStd_Document:
+    """
+    Converts the assembly to a fused compound and saves that within the document
+    to be exported in a way that preserves the face colors. Because of the use of
+    boolean operations in this method, performance may be slow in some cases.
+
+    :param assy: Assembly that is being converted to a fused compound for the document.
+    :param assy_name: Label that is applied to the top level object in the document.
+    """
+
+    # Prepare the document
+    app = XCAFApp_Application.GetApplication_s()
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+    app.InitDocument(doc)
+
+    # Shape and color tools
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+
+    # To fuse the parts of the assembly together
+    fuse_op = BRepAlgoAPI_Fuse()
+    args = TopTools_ListOfShape()
+    tools = TopTools_ListOfShape()
+
+    # If there is only one solid, there is no reason to fuse, and it will likely cause problems anyway
+    top_level_shape = None
+    children = cast(List, assy.children)
```

Comment:
That cast should not be needed

Reply:
Without it mypy shows the following errors.
```
cadquery/occ_impl/assembly.py:381: error: Value of type "Iterable[AssemblyProtocol]" is not indexable  [index]
cadquery/occ_impl/assembly.py:382: error: Value of type "Iterable[AssemblyProtocol]" is not indexable  [index]
cadquery/occ_impl/assembly.py:385: error: Value of type "Iterable[AssemblyProtocol]" is not indexable  [index]
```